tests/functional/adapter/test_override_database/test_override_database.py

```diff
@@ -0,0 +1,158 @@
+import pytest
+from dbt.tests.util import run_dbt, check_relations_equal, check_relations_equal_with_relations
+from tests.functional.adapter.test_override_database.fixtures import (
+    models,
+    seeds,
+    project_files
+)
+import os
+
+
+
+
+class BaseOverrideDatabase:
+    @pytest.fixture(scope="class")
+    def model_path(self):
+        return "models"
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "seed-paths": ["seeds"],
+            "vars": {
+                "alternate_db": os.getenv("BIGQUERY_TEST_ALT_DATABASE"),
+            },
+            "quoting": {
+                "database": True,
+            },
+            "seeds": {
+                "quote_columns": False,
+            }
+        }
+
+
+class TestModelOverrideBigQuery(BaseOverrideDatabase):
+    def run_database_override(self, project):
+        run_dbt(["seed"])
+        assert len(run_dbt(["run"])) == 4
+        check_relations_equal_with_relations(project.adapter, [
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_DATABASE"), schema=project.test_schema, identifier="seed"),
```

Comment:
We could either:
- Replace `database=os.getenv("BIGQUERY_TEST_DATABASE")` with `database=project.database`
- Remove the `database` argument entirely, and fix a v1.1 regression (https://github.com/dbt-labs/dbt-bigquery/issues/180) by restoring support for using default `database`/`schema` when unspecified in `get_columns_in_relation`